code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -75,7 +74,7 @@ public Account getAccountByExternalId(String externalId, String externalType) {
                 .where(
                         ACCOUNT.EXTERNAL_ID.eq(externalId)
                         .and(ACCOUNT.EXTERNAL_ID_TYPE.eq(externalType))
-                        .and(ACCOUNT.STATE.eq("active"))
+                        .and(ACCOUNT.STATE.ne("purged"))
```

Comment:
This is used by `io.cattle.platform.iaas.api.auth.github.GithubTokenHandler.getGithubAccessToken(ApiRequest)`  is this really a safe change?